core/model/modx/moduser.class.php

```diff
@@ -787,7 +787,10 @@ public function removeLocks(array $options = array()) {
      * @param array $options
      * @return string The newly generated password
      */
-    public function generatePassword($length = 10,array $options = array()) {
+    public function generatePassword($length = null,array $options = array()) {
+        if ($length === null) {
+            $length = $this->getOption('password_generated_length',null,10);
```

Comment:
@JoshuaLuckers Just a comment, not requirement. But as we are fixing this part maybe we should use here length that set in system settings by default to keep consistency? In system settings, it set as 8 symbols.

Reply:
Another issue that I found it when to set `password_generated_length` as an empty string (not null, but '') generator will return the empty password. As I remember getOption does not check empty strings and return it as is. And we need to check it additionally.
At least we have inconsistency between `getOption` in `xPDOObject` and in `xPDO` classes. The second one has fourth parameter `$skipEmpty = false` that allows properly check empty strings when it set in true. 
Details here https://github.com/modxcms/revolution/blob/cdbbb8f611ce2b7c6e6460895b297ed5cdf54c75/core/xpdo/xpdo.class.php#L686 and https://github.com/modxcms/revolution/blob/31fdf1a342844f9efc854e9d6f7c18815e28c09e/core/xpdo/om/xpdoobject.class.php#L697

Reply:
Thanks! I decided to use 10 because the current method signature also uses 10. I would suggest increasing the default system settings to 10 instead.